docs: document passkey advanced configuration options (#2519)
ory/docs: docs/kratos/passwordless/05_passkeys.mdx

</Tabs>
```

### Advanced configuration

> The passkey strategy exposes additional options that control the WebAuthn ceremony. All options are optional and existing passkey

Reviewer: ```suggestion
The passkey strategy exposes additional settings that control the WebAuthn ceremony. All are optional and existing passkey
```

Author: 👍

> The passkey strategy exposes additional options that control the WebAuthn ceremony. All options are optional and existing passkey

Reviewer: The Console UI says that 'Changing these settings below might block existing users from logging in.'

[ConsoleUI.pdf](https://github.com/user-attachments/files/28659129/ConsoleUI.pdf)

Author: Yeah thats a good safety note, but in this case it should be safe. Lets error on the side of caution though.

| ------------------------------------------- | -------- | --------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------ |
| `authenticator_selection.attachment`        | string   | _unset_         | Restricts eligible authenticators by attachment modality: `"platform"` (Touch ID, Windows Hello) or `"cross-platform"` (HID tokens, YubiKeys). Omit this field to accept either type.                              |
| `authenticator_selection.resident_key`      | string   | `"required"`    | Whether the authenticator must create a client-side discoverable credential: `"required"`, `"preferred"`, or `"discouraged"`.                                                                                      |
| `authenticator_selection.user_verification` | string   | `"preferred"`   | Whether biometrics or a PIN are required: `"required"`, `"preferred"`, or `"discouraged"`.                                                                                                                         |

> | `attestation.preference`                    | string   | `"none"`        | Attestation conveyance preference Kratos _requests_ from the authenticator: `"none"`, `"indirect"`, `"direct"`, or `"enterprise"`. See the note below — this is a request hint only, not an enforcement control.   |

Reviewer: ```suggestion
| `attestation.preference`                    | string   | `"none"`        | Attestation conveyance preference that Ory Kratos _requests_ from the authenticator: `"none"`, `"indirect"`, `"direct"`, or `"enterprise"`. See the note below — this is a request hint only, not an enforcement control.   |
```

Author: Fixed.

| `timeouts.registration`                     | duration | library default | Timeout for the registration ceremony. Use Go duration format, for example `"60s"` or `"5m"`. When unset, the library default applies: 5 minutes, or 2 minutes when `user_verification` is set to `"discouraged"`. |

> | `timeouts.login`                            | duration | library default | Timeout for the login ceremony. Use Go duration format. When unset, the same default behavior as `timeouts.registration` applies.                                                                                  |

Reviewer: ```suggestion
| `timeouts.login`                            | duration | library default | Timeout for the login ceremony. Use Go duration format. When unset, the library default applies: 5 minutes, or 2 minutes when `user_verification` applies.                                                                                  |
```

Author: Addressed.

:::note

**Default behavior change:** Earlier versions of the passkey strategy hardcoded `attachment: platform`, so only built-in
authenticators (Touch ID, Windows Hello) could be registered. The strategy now defaults to **no attachment preference**, which

> means users can register either platform or cross-platform (roaming) authenticators. To restore the previous behavior, set

Reviewer: ```suggestion
means users can register using either platform or cross-platform (roaming) authenticators. To restore the previous behavior, set
```

`authenticator_selection.attachment` to `"platform"`.

:::

:::note

**Timeouts are enforced by the browser:** `timeouts.registration` and `timeouts.login` are communicated to the browser, which

> enforces them during the WebAuthn ceremony. Kratos does not reject late responses server-side based on these values — the login

Reviewer: ```suggestion
enforces them during the WebAuthn ceremony. Ory Kratos does not reject late responses server-side based on these values — the login
```

Author: Resolved

> enforces them during the WebAuthn ceremony. Kratos does not reject late responses server-side based on these values — the login

Reviewer: Why a note? Can we fold this info into the table?

Author: Got it, sorted.

and registration flow lifespans (`selfservice.flows.*.lifespan`) govern how long a flow remains valid.

:::

:::note

> 

Reviewer: Why a note? Can you fold this info into the table?

Author: much cleaner, did that!

> **`attestation.preference` is request-only:** This option controls only what Kratos requests from the authenticator. Kratos does

Reviewer: ```suggestion
**`attestation.preference` is request-only:** This option controls only what Ory Kratos requests from the authenticator. Ory Kratos does
```

Author: Fixed.
